File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/java/robots/MenuRobot.java
package pavelnazimok.uitestingfeatures.java.robots;

import pavelnazimok.uitestingfeatures.R;

import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;

@SuppressWarnings("UnusedReturnValue")
public class MenuRobot {

    public MenuRobot clickFirstScreen() {
        clickOn(R.id.nav_first);
        return this;
    }

    public MenuRobot clickSecondScreen() {
        clickOn(R.id.nav_second);
        return this;
    }

    public MenuRobot clickThirdScreen() {
        clickOn(R.id.nav_third);
        return this;
    }
}

File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/kotlin/robots/MenuRobot.kt
package pavelnazimok.uitestingfeatures.kotlin.robots

import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn
import pavelnazimok.uitestingfeatures.R
import pavelnazimok.uitestingfeatures.kotlin.utils.waitForView

fun onMenu(func: MenuRobot.() -> Unit): MenuRobot {
    return MenuRobot().apply {
        assertMenuIsDisplayed()
        func()
    }
}

class MenuRobot {

    fun clickFirstScreen() {
        clickOn(R.id.nav_first)
    }

    fun clickSecondScreen() {
        clickOn(R.id.nav_second)
    }

    fun clickThirdScreen() {
        clickOn(R.id.nav_third)
    }

    fun assertMenuIsDisplayed() {
        waitForView(R.id.nav_bottom_bar)
    }
}